web/src/components/ChatConversation.tsx

```diff
@@ -37,6 +37,14 @@ type TypingIndicatorProps = {
   isVisible: boolean
 }
 
+const shouldShowIcon = (index: number, message: ChatMessageModel, messages: ChatMessageModel[]): boolean => {
+  const previousMessage = messages[index - 1]
+  return (
+    previousMessage?.userIsAuthor !== message.userIsAuthor ||
+    previousMessage.isAutomaticAnswer !== message.isAutomaticAnswer
+  )
+}
```

Comment:
:wrench: Perhaps a cleaner solution would be to pass `previousMessage` to `ChatMessage` and move this logic to decide whether to display the icon or not there.